addons/mail/static/src/views/fields/statusbar_duration/statusbar_duration_field.js

```diff
@@ -0,0 +1,114 @@
+        if (seconds < 60) {
+            seconds = 60;
+        }
+        const minuteInSeconds = 60;
+        const hourInSeconds = 60 * minuteInSeconds;
+        const dayInSeconds = 24 * hourInSeconds;
+        const monthInSeconds = 30 * dayInSeconds;
+        const yearInSeconds = 365 * dayInSeconds;
+        const numberOfValues = showFullDuration ? 2 : 1;
+        const displayStyle = showFullDuration ? "long" : "narrow";
+
+        const years = Math.floor(seconds / yearInSeconds);
+        seconds -= years * yearInSeconds;
+        const months = Math.floor(seconds / monthInSeconds);
+        seconds -= months * monthInSeconds;
+        const days = Math.floor(seconds / dayInSeconds);
+        seconds -= days * dayInSeconds;
+        const hours = Math.floor(seconds / hourInSeconds);
+        seconds -= hours * hourInSeconds;
+        const minutes = Math.floor(seconds / minuteInSeconds);
+        seconds -= minutes * minuteInSeconds;
+        const durationObject = {};
+        if (years) {
+            durationObject.years = years;
+        }
+        if (months && Object.keys(durationObject).length < numberOfValues) {
+            durationObject.months = months;
+        }
+        if (days && Object.keys(durationObject).length < numberOfValues) {
+            durationObject.days = days;
+        }
+        if (hours && Object.keys(durationObject).length < numberOfValues) {
+            durationObject.hours = hours;
+        }
+        if (minutes && Object.keys(durationObject).length < numberOfValues) {
+            durationObject.minutes = minutes;
+        }
+        if (Object.keys(durationObject).length === 0) {
+            return false;
+        }
+        const duration = luxon.Duration.fromObject(durationObject);
+        let durationStr = duration.toHuman({ unitDisplay: displayStyle });
+
+        // In English, the symbol for minutes and months is the same (m), so change months to M
+        if (!showFullDuration && duration.loc.locale.includes("en") && durationObject.months > 0) {
+            const index = durationStr.indexOf("m");
+            durationStr =
+                durationStr.substring(0, index) +
+                "M" +
+                durationStr.substring(index + 1, durationStr.length);
+        }
+        return durationStr;
+    }
+
+    setStatusbarDuration(stages) {
+        const durationTracking = this.props.record.data.duration_tracking;
+        if (!Object.keys(durationTracking).length) {
+            return;
+        }
+        for (const stage of stages) {
+            if (stage.id in durationTracking && durationTracking[stage.id] > 0) {
+                stage.shortTimeInStage = this.humanizeDuration(durationTracking[stage.id], false);
+                stage.fullTimeInStage = this.humanizeDuration(durationTracking[stage.id], true);
+            } else {
+                stage.shortTimeInStage = false;
+            }
+        }
+    }
+
+    computeItems(grouped = true) {
```

Comment:
Beware of alphabetical ordering (in JS it is a guideline, at least inside "sections" (public, private, ...))

Reply:
Done